crates/common/tedge_config/src/tedge_toml/figment.rs

```diff
@@ -269,6 +340,90 @@ mod tests {
         })
     }
 
+    #[test]
+    fn integer_environment_variables_are_parsed_as_integers() {
+        #[derive(Deserialize)]
+        struct Config {
+            value: u32,
+        }
+
+        figment::Jail::expect_with(|jail| {
+            jail.set_env("TEDGE_VALUE", "1234");
+
+            assert_eq!(
+                extract_data::<Config, FileAndEnvironment>(&PathBuf::from("tedge.toml"))
+                    .unwrap()
+                    .0
+                    .value,
+                1234
+            );
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn environment_variables_with_leading_zeroes_are_parsed_as_strings() {
+        #[derive(Deserialize)]
+        struct Config {
+            value: String,
+        }
+
+        figment::Jail::expect_with(|jail| {
+            jail.set_env("TEDGE_VALUE", "01234");
+
+            assert_eq!(
+                extract_data::<Config, FileAndEnvironment>(&PathBuf::from("tedge.toml"))
+                    .unwrap()
+                    .0
+                    .value,
+                "01234",
+            );
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn environment_variable_zero_is_parsed_as_number() {
+        #[derive(Deserialize)]
+        struct Config {
+            value: u32,
+        }
+
+        figment::Jail::expect_with(|jail| {
+            jail.set_env("TEDGE_VALUE", "0");
+
+            assert_eq!(
+                extract_data::<Config, FileAndEnvironment>(&PathBuf::from("tedge.toml"))
+                    .unwrap()
+                    .0
+                    .value,
+                0,
+            );
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn environment_variable_float_is_parsed_as_number() {
+        #[derive(Deserialize)]
+        struct Config {
+            value: f64,
+        }
+
+        figment::Jail::expect_with(|jail| {
+            jail.set_env("TEDGE_VALUE", "0.123");
+
+            assert_eq!(
+                extract_data::<Config, FileAndEnvironment>(&PathBuf::from("tedge.toml"))
+                    .unwrap()
+                    .0
+                    .value,
+                0.123,
+            );
+            Ok(())
+        })
+    }
+
```

Comment:
suggestion: Would it be possible to have a test when we deserialize to both a number and to a string from all the 3 cases from my comment: https://github.com/thin-edge/thin-edge.io/pull/3516#issuecomment-2761494136 ?

The changes look roughly correct, but I still have some trouble understanding the implication of this change for consumers that deserialize sequences of digits into strings (`device.cryptoki.pin`) or into proper numbers.
> Technically, this could be seen as a breaking change, but in a practical sense it shouldn't be
This is one question I had, does this change result in a thing that was previously possible now being an error? It would be nice to have a confirmation, if any, in a form of a test.